breadth_first_search/BreadthFirstSearch.java

```diff
@@ -0,0 +1,65 @@
+import java.util.*;
+
+public class BreadthFirstSearch <T> {
+
+    // HasMap  of lists for Adjacency List Representation
+    public HashMap<Integer, ArrayList<T>> adj;
+
+    //Function to add an edge into the breadthFirstTraversal
+    private void addEdge(int source, T destination) {
+        adj.get (source).add (destination);
+    }
+
+    public ArrayList breadthFirstSearch(T source, T destination) {
+        // false by default in java)
+        ArrayList arrayList = new ArrayList();
+        Set<T> visited = new HashSet();
+        // Mark the current node as visited
+        visited.add(source);
+        ArrayList<T> queue = new ArrayList<> ();
+        queue.add(source);
+        while (queue.size() != 0) {
+            source = queue.remove(0);
+            arrayList.add(source);
+            Iterator<T> i = adj.get(source).listIterator();
```

Comment:
The index is the type of integer in the hash map and ArrayList data type of T if I do not wrong.
Please @aviaryan check it.